### ISSUE TITLE
[Integration]: Setup MLflow Experiment Tracking

### ISSUE THREAD

**Issue by spencerduncan** [spencerduncan/jimbot#33]
## Description
Integrate MLflow for tracking machine learning experiments, model performance, and strategy evolution in the Balatro learning system.

## Acceptance Criteria
- [ ] MLflow server deployed and configured
- [ ] Experiment tracking API integrated
- [ ] Model registry operational
- [ ] Metrics logging implemented
- [ ] Artifact storage configured
- [ ] UI accessible for experiment review

### MLflow Setup:
```python
import mlflow
from mlflow.tracking import MlflowClient

class ExperimentTracker:
    def __init__(self):
        mlflow.set_tracking_uri("http://localhost:5000")
        self.client = MlflowClient()
        
    def log_game_metrics(self, session_id: str, metrics: dict):
        with mlflow.start_run(run_name=f"game_{session_id}"):
            mlflow.log_metrics(metrics)
            mlflow.log_param("strategy", metrics.get("strategy_id"))
            
    def log_model_performance(self, model_version: str, performance: dict):
        mlflow.log_metric("win_rate", performance["win_rate"])
        mlflow.log_metric("avg_score", performance["avg_score"])
        mlflow.log_metric("learning_rate", performance["learning_rate"])
```

### Metrics to Track:
- Strategy performance (win rate, scores)
- Learning progress (loss, accuracy)
- Hyperparameters (learning rate, batch size)
- Resource usage (GPU, memory)
- Decision quality metrics

## Dependencies
- **Blocked by**: #17 (Event Bus Consumer)
- **Blocks**: Advanced analytics
- **Parent**: #8

## Implementation Notes
- Use PostgreSQL backend for MLflow
- Configure S3-compatible storage for artifacts
- Set up automatic experiment organization
- Implement model versioning
- Add comparison capabilities

Integration points:
- Learning orchestration logs experiments
- Analytics queries MLflow for insights
- Dashboards display experiment metrics

## Definition of Done
- [ ] MLflow server running
- [ ] Tracking API integrated
- [ ] Sample experiments logged
- [ ] UI accessible and functional
- [ ] Backup procedures defined
- [ ] Documentation complete

**Comment by spencerduncan** [comment]
**Blocked by**: #17 (Event Bus Consumer)
**Blocks**: Advanced analytics features

**Comment by spencerduncan** [comment]
This issue is now obsolete. The project has chosen to use Ray RLlib for reinforcement learning tracking instead of MLflow. Ray RLlib provides integrated experiment tracking and model management that better aligns with our distributed training architecture.
